### ISSUE TITLE
Remove dependency on toolbar from clipboard

### ISSUE THREAD

**Issue by Comandeer** [ckeditor/ckeditor-dev#654]
## Are you reporting a feature or a bug?

Task

## Explanation

Currently clipboard plugin depends on two other plugins: notification and toolbar. However it seems that toolbar-specific functions are used only to disallow dropping files on it. Therefore dropping it as a dependency seems like a relatively simple task.

It's kinda-sorta followup of #491.

**Comment by mlewand** [comment]
Definitely -1 for depending on toolbar in clipboard plugin.

**Comment by jonnott** [comment]
Does the merging of https://github.com/ckeditor/ckeditor-dev/pull/631 mean this can be closed now?

**Comment by Comandeer** [comment]
Actually #631 introduced explicit `toolbar` dependency, so unfortunately not – it has to be resolved separately.

**Comment by jonnott** [comment]
@Comandeer can you please explain "introduced explicit toolbar dependency" - are we in a worse state than we started now?

**Comment by Comandeer** [comment]
No, it's the same state.

Before merging #631 `clipboard` depended on `notification` plugin, which depended on `toolbar` plugin. Thanks to that, `clipboard` could use `toolbar` functions without explicitly depending on it (due to the rule "dependencies of my dependencies are my dependencies").

After merging #631 `clipboard` still uses some functions from `toolbar`, but due to the fact that `notification` does not depend on it anymore, `clipboard` must now depend on `notification` and `toolbar`.

**Comment by jonnott** [comment]
@Comandeer This is ridiculous, as it's just moving the problem elsewhere. You should be able to:

removePlugins: 'toolbar'

..and any other plugins that use the toolbar plugin should adapt around this.

What's the point in having the ability to remove the toolbar at all otherwise? And what's the point of these issues/PRs!!

**Comment by Comandeer** [comment]
The previous issue was connected with `notification` plugin and its purpose was to remove `toolbar` dependency from `notification` – and that purpose was fulfilled. This issue is taking the whole process one step further and aims to remove this dependency also from `clipboard` plugin. After resolving this issue removing toolbar should be possible.

**Comment by jonnott** [comment]
@Comandeer That would be great .. I really would love to be able to simply do

removePlugins: 'toolbar'

when I don't want to have a toolbar!

**Comment by jonnott** [comment]
Can this be tagged for the 4.7.2 milestone?

**Comment by Comandeer** [comment]
Unfortunately this issue is blocked on #678, so it would be postponed to 4.8.0 release.

**Comment by mlewand** [comment]
Unfortunately, we need to postpone it together with #678 to 4.9.0 release. This is an important feature for us, but we can not block delay 4.8.0 release any further.

**Comment by jonnott** [comment]
Why have this issue, and #678, both been removed from 4.9.0 milestone? @mlewand @Comandeer

**Comment by jonnott** [comment]
Still no headway with this cluster of related issues in v4.10 ... ???? When will it be addressed, it seems to affect so many things!

**Comment by jestevesRedeye** [comment]
Hello, can find this issue in the release 4.14.0, and investigation lead me to this page.
Any work around to hide the inline toolbar? (apart from CSS)
We are moving from 4.6.2 as we are aware of XSS vulnerability. Thanks
